### PR TITLE
Adding Front Proxy article

### PR DESCRIPTION
**Pull request by brookshelley**
New front proxy article for Learn Envoy. Added to the Deployment Models section.

### CODE REVIEW

File: _articles/front-proxy.md [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+these to a set of Envoy routes. See
+[Routing Configuration](https://www.learnenvoy.io/articles/routing-configuration.html) for more details.
+
+Most modern apps will want to implement a dynamic control plane, since the
+instances within each service are dynamic. A static configuration can only
+point to a fixed set of instances, while a dynamic control plane can keep Envoy
+up-to-date on the state of your environment, typically by reading from your
+service discovery registry. See
+[Integrating Service Discovery with Envoy](https://www.learnenvoy.io/articles/service-discovery.html)
+for implementations that will do this for you, like [Rotor](https://github.com/turbinelabs/rotor).
```
Comment (trjordan) [inline _articles/front-proxy.md]: wrap

File: _articles/front-proxy.md [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+unit, but cross-zone traffic incurs a cost. In this setup, your network load
+balancer would balance across a front proxy that spans multiple zones, and
+Envoy would keep each request in the zone it was first load balanced to. This
+will minimize cross-zone traffic (which is expensive and slow), while
+maintaining the ability to incrementally fall back to out-of-zone instances if
+the local ones are failing.
+
+Setting up zone-aware load balancing requires two options to be set:
+
+  - Each cluster (statically defined or returned by the control plane via CDS) must enable zone-aware routing.
```
Comment (trjordan) [inline _articles/front-proxy.md]: Wrap, and also does this render correctly? Not sure about the space on the next line properly translating to a list.

File: _articles/front-proxy.md [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+or a minimal configuration of a more feature-rich load balancer
+like AWS ALB. These baked-in load balancers can handle regional or global
+traffic balancing, which is impossible for a set of VMs running Envoy (which
+are necessarily constrained to a single availability zone) to handle.
+
+A simple, robust deployment of an Envoy front proxy uses an autoscaling group
+of Envoys based on network traffic. Based on where your routing rules currently
+live (NGINX config files, AWS ALB configuration, etc.), you will need to port
+these to a set of Envoy routes. See
+[Routing Configuration](https://www.learnenvoy.io/articles/routing-configuration.html) for more details.
```
Comment (trjordan) [inline _articles/front-proxy.md]: wrap

File: _articles/front-proxy.md [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: envoy-front-proxy
+spec:
+  replicas: 2
+  template:
+    spec:
+      containers:
+      - image: turbinelabs/envoy-simple:0.16.0
```
Comment (trjordan) [inline _articles/front-proxy.md]: Bump to current version, 0.17.2

File: _articles/front-proxy.md [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+## Next Steps
+
+While this article has focused on how to handle traffic coming from outside
+your network, it's also possible for Envoy to handle traffic between services
+(or “east-west” traffic). For a lightweight way to set up this internal mesh,
+you can route internal requests through this front proxy (or a similarly
+configured proxy pool specifically for east-west).
+
+Beyond that, you can take better advantage of Envoy’s unique features as a
+lightweight sidecar by setting up a Basic Service Mesh.
```
Comment (trjordan) [inline _articles/front-proxy.md]: Maybe add a "coming soon!" since we haven't published this article yet?

File: _articles/front-proxy.md [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+
+This means [retries](automatic-retries.html),
+[health checks](health-check.html), etc. What this looks like depends strongly
+on your infrastructure and the types of issues you’re looking to mitigate.
+
+## Next Steps
+
+While this article has focused on how to handle traffic coming from outside
+your network, it's also possible for Envoy to handle traffic between services
+(or “east-west” traffic). For a lightweight way to set up this internal mesh,
```
Comment (trjordan) [inline _articles/front-proxy.md]: I think we have problems with smart quotes in markdown?

File: _articles/front-proxy.md (hunk 3 above)
Comment (trjordan) [reply _articles/front-proxy.md]: also would it make more sense to do relative links for internal links? not sure it matters, though we do it below for SSL / retries / health checks

File: _articles/front-proxy.md (hunk 2 above)
Comment (brookshelley) [reply _articles/front-proxy.md]: yea it works ok

File: _articles/front-proxy.md (hunk 3 above)
Comment (brookshelley) [reply _articles/front-proxy.md]: yea I can do relatives

File: _articles/front-proxy.md (hunk 5 above)
Comment (brookshelley) [reply _articles/front-proxy.md]: gonna redact and add this back in after service mesh publishes

**Comment by brookshelley** [comment]
🎱

File: _articles/front-proxy.md [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+live (NGINX config files, AWS ALB configuration, etc.), you will need to port
+these to a set of Envoy routes. See
+[Routing Configuration](outing-configuration.html) for more details.
+
+Most modern apps will want to implement a dynamic control plane, since the
+instances within each service are dynamic. A static configuration can only
+point to a fixed set of instances, while a dynamic control plane can keep Envoy
+up-to-date on the state of your environment, typically by reading from your
+service discovery registry. See
+[Integrating Service Discovery with Envoy](https://www.learnenvoy.io/articles/service-discovery.html)
```
Comment (trjordan) [inline _articles/front-proxy.md]: Relative link?

**Pull request merged**